File: src/C3.Catalogue/CassetteModels/CassetteModelDraft.vb
Namespace CassetteModels

    Public NotInheritable Class CassetteModelDraft

        Public Sub New(
                brandCode As String,
                typeNumber As Integer,
                modelName As String,
                code As String,
                displayName As String,
                notes As String)

            Me.BrandCode = brandCode
            Me.TypeNumber = typeNumber
            Me.ModelName = modelName
            Me.Code = code
            Me.DisplayName = displayName
            Me.Notes = notes
        End Sub

        Public ReadOnly Property BrandCode As String
        Public ReadOnly Property TypeNumber As Integer
        Public ReadOnly Property ModelName As String
        Public ReadOnly Property Code As String
        Public ReadOnly Property DisplayName As String
        Public ReadOnly Property Notes As String

    End Class

End Namespace

File: src/C3.Catalogue/CassetteModels/CassetteModelDraft.cs
namespace C3.Catalogue.CassetteModels
{
    public sealed class CassetteModelDraft
    {
        public CassetteModelDraft(
            string brandCode,
            int typeNumber,
            string modelName,
            string code,
            string displayName,
            string notes)
        {
            BrandCode = brandCode;
            TypeNumber = typeNumber;
            ModelName = modelName;
            Code = code;
            DisplayName = displayName;
            Notes = notes;
        }

        public string BrandCode { get; }

        public int TypeNumber { get; }

        public string ModelName { get; }

        public string Code { get; }

        public string DisplayName { get; }

        public string Notes { get; }
    }
}
